ee/tabby-webserver/src/service/background_job/delete_documents.rs

```diff
@@ -0,0 +1,30 @@
+use std::sync::Arc;
+
+use tabby_inference::Embedding;
+use tabby_scheduler::DocIndexer;
+use tabby_schema::Result;
+use tracing::debug;
+
+use super::helper::Job;
+
+pub struct DeleteIndexedDocumentsJob {
+    source: String,
+}
+
+impl DeleteIndexedDocumentsJob {
+    pub fn new(source: String) -> Self {
+        Self { source }
+    }
+
+    pub async fn run(self, embedding: Arc<dyn Embedding>) -> Result<()> {
```

Comment:
delete job shouldn't really need embedding - consider pass in a void implementation or refactor DocIndexer to take Option embedding input.